src/vivarium_gates_nutrition_optimization/components/children.py

```diff
@@ -0,0 +1,72 @@
+from pathlib import Path
+
+import pandas as pd
+from vivarium.framework.engine import Builder
+from vivarium.framework.event import Event
+from vivarium_cluster_tools.utilities import mkdir
+
+from vivarium_gates_nutrition_optimization.constants import models
+
+
+class BirthRecorder:
+    @property
+    def name(self):
+        return "birth_recorder"
+
+    #################
+    # Setup methods #
+    #################
+
+    # noinspection PyAttributeOutsideInit
+    def setup(self, builder: Builder) -> None:
+        self.output_path = self._build_output_path(builder)
+        self.randomness = builder.randomness.get_stream(self.name)
+
+        self.births = []
+
+        required_columns = [
+            "pregnant_event_time",
+            "pregnancy_term_outcome",
+            "pregnancy_duration",
+            "pregnancy",
+            "previous_pregnancy",
+        ]
+        self.population_view = builder.population.get_view(required_columns)
+
+        builder.event.register_listener("collect_metrics", self.on_collect_metrics)
+        builder.event.register_listener("simulation_end", self.write_output)
+
+    def on_collect_metrics(self, event: Event):
+        pop = self.population_view.get(event.index)
+        new_birth_mask = (
+            (pop["pregnancy_term_outcome"] == models.FULL_TERM_OUTCOME)
+            & (pop["previous_pregnancy"] == models.PREGNANT_STATE_NAME)
+            & (pop["pregnancy"] == models.POSTPARTUM_STATE_NAME)
+        )
+
+        pop["birth_date"] = pop["pregnant_event_time"] + pop["pregnancy_duration"]
```

Comment:
We don't actually want the birth date. What we really want is the pregnancy duration, since in the child simulation all the children will be born at the beginning of the simulation. In the previous MNCH model the children were born at different times throughout the simulation, so we needed the birth date. 

This also means we don't need the `"pregnant_event_time"` column.